weave-js/src/common/components/elements/ModifiedDropdown.tsx

```diff
@@ -482,12 +482,15 @@ const OptionWithTooltip: React.FC<OptionWithTooltipProps> = ({text}) => {
   const optionRef = useRef<HTMLDivElement>(null);
 
   useEffect(() => {
-    if (optionRef.current) {
-      setShowTooltip(
-        optionRef.current.scrollWidth > optionRef.current.clientWidth
-      );
+    if (!optionRef.current) {
+      return;
+    }
+    const isOverflow =
+      optionRef.current.scrollWidth > optionRef.current.clientWidth;
+    if (isOverflow !== showTooltip) {
+      setShowTooltip(isOverflow);
```

Comment:
could this just be `setShowTooltip(isOverflow);`? React shouldn't cause an update if it ends up the same boolean value 🤔

Reply:
oh wait duh 😂 , that was the previous code. I wonder if this is because of the ref